syntax = "proto3";

package attendance;

service AttendanceService {
    rpc CreateNotification(CreateNotificationRequest) returns (NotificationResponse);
    rpc GetUserNotifications(GetUserNotificationsRequest) returns (UserNotificationsResponse);
    rpc MarkAttendance(MarkAttendanceRequest) returns (SuccessResponse);
    rpc GetAttendance(GetAttendanceRequest) returns (AttendanceResponse);
}

message CreateNotificationRequest {
    string user_id = 1;
    string type = 2;
    string title = 3;
    string message = 4;
    optional string related_booking_id = 5;
    optional string related_workout_id = 6;
}

message GetUserNotificationsRequest {
    string user_id = 1;
    optional int32 limit = 2;
}

message MarkAttendanceRequest {
    string user_id = 1;
    string workout_id = 2;
    bool attended = 3;
}

message GetAttendanceRequest {
    string workout_id = 1;
}

message NotificationResponse {
    string notification_id = 1;
    string user_id = 2;
    string type = 3;
    string title = 4;
    string message = 5;
    string created_at = 6;
    optional string related_booking_id = 7;
    optional string related_workout_id = 8;
}

message UserNotificationsResponse {
    repeated NotificationResponse notifications = 1;
}

message AttendanceResponse {
    repeated AttendanceRecord attendance = 1;
    int32 total = 2;
    int32 attended = 3;
}

message AttendanceRecord {
    string user_id = 1;
    string workout_id = 2;
    bool attended = 3;
    string checked_in_at = 4;
}

message SuccessResponse {
    bool success = 1;
    string message = 2;
}

message Empty {}